### DISCUSSION TITLE
How to authenticate users with credentials with the Firebase adapter?

### DISCUSSION THREAD

**Discussion by IDrumsey** [nextauthjs/next-auth#5526] [Help]
### Question 💬

Hello, I'm trying to authenticate users with the Firebase adapter using email/password. I followed [this](https://next-auth.js.org/adapters/firebase#getting-started) guide to attach the Firebase adapter. The social provider I have attached works as expected with the adapter. Now I'm trying to add email/password authentication. As per [this doc](https://next-auth.js.org/providers/), there are four provider options. I'm assuming I need to use the Credentials provider ([see this](https://next-auth.js.org/configuration/providers/credentials#:~:text=The%20Credentials%20provider%20allows%20you%20to%20handle%20signing%20in%20with%20arbitrary%20credentials%2C%20such%20as%20a%20username%20and%20password)). So I tried with a simple authentication function in the Credentials provider, but when I go to sign in with the provider I get directed back to the NEXTAUTH_URL without even seeing the form.

```javascript
// in providers array
CredentialsProvider({
            credentials: {
                username: { label: "username", type: "text", placeholder: "your username" },
                password: { label: "password", type: "password" }
            },

            async authorize(creds, req) {
                return {
                    id: 1,
                    username: "John"
                }

                return null
            }
        })
```

Am I on the right track using the Credentials provider?

Or do I maybe need to use the firebase sdk to do this? I was looking at the [code](https://github.com/nextauthjs/next-auth/blob/d13997e1400af0f5d9cce286750414b85a91dbea/packages/adapter-firebase/src/index.ts#L66-L96) and it looks like there are already methods for doing this kind of thing, so I'd assume I don't have to use the sdk. It looks like there was an [update](https://github.com/nextauthjs/next-auth/pull/3873) to the adapter earlier this summer.

Any help is appreciated, thank you :)

### How to reproduce ☕️

Write the [...nextauth].js file with the Credentials provider and Firebase adapter.

Then use the signin function as normal.

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by vecode** [comment]
I'm also trying to set this up and would appreciate any help.
